js/stores/windowStore.js

```diff
@@ -398,21 +397,8 @@ const doAction = (action) => {
         }
         break
       }
-    case windowConstants.WINDOW_TAB_MOUSE_MOVE:
-      {
-        // previewMode is only triggered if mouse is idle over a tab
-        // for a given amount of time based on timing defined in prefs->tabs
-        // we use actions here because that is the only way to delay updating the state
-        clearTimeout(mouseTimeout)
```

Comment:
we specifically wanted to keep the timeout in the action handler vs putting it in the component

Reply:
can you please elaborate a little bit more. I don't see a real benefit and we are reducing performance when calling action without state change. Because mouse move is happening on a tab and this functionality is limited to this component, I don't see why we can't have it inside the component.

Reply:
the benefit is that it keeps the hover logic out of the components. The action shouldn't be an expensive call and is likely dwarfed by the relatively expensive `setTimeout` call. Have you measure the performance difference between the two of them?

Reply:
I'm not 100% opposed to this change, but I'd like to see some hard perf numbers first

Reply:
@bridiver here are some numbers https://github.com/brave/browser-laptop/pull/10690#issuecomment-325886207

Reply:
My 2c is that it's actually cleaner (as well as more performant) to put this logic in the component - the UI component is deciding when the right UI actions have been performed by the user and the action should be sent, not the store.

Reply:
I disagree that it's cleaner. First, we're back to an imperative action and those have been the source of endless problems. The component is also now deciding when the state should change, not when the UI action has been performed. The UI action is a mouse move. The state change for hover state may or may not occur depending based on that mouse move and conceivably something else might care about that mouse move without the timeout.